pkg/agent/manager/storage.go

```diff
@@ -61,11 +61,8 @@ func ReadSVID(svidCachePath string) (*x509.Certificate, error) {
 	return cert, nil
 }
 
-// storeSVID writes manager's svid to disk into SVIDCachePath. Returns nil if all went
+// StoreSVID writes the specified svid to disk into svidCachePath. Returns nil if all went
 // fine, otherwise ir returns an error.
-func (m *manager) storeSVID() error {
-	m.mtx.RLock()
-	data := m.svid.Raw
-	m.mtx.RUnlock()
-	return ioutil.WriteFile(m.svidCachePath, data, 0600)
+func StoreSVID(svidCachePath string, svid *x509.Certificate) error {
+	return ioutil.WriteFile(svidCachePath, svid.Raw, 0600)
```

Comment:
at some point we might want to consider the writing to a temp file and then renaming over the top.